internal/configuration/settings/serverselection.go

```diff
@@ -107,6 +123,11 @@ func (ss *ServerSelection) validate(vpnServiceProvider string,
 		return fmt.Errorf("for VPN service provider %s: %w", vpnServiceProvider, err)
 	}
 
+	err = validateFeatureFilters(*ss, vpnServiceProvider)
+	if err != nil {
+		return fmt.Errorf("for VPN service provider %s: %w", vpnServiceProvider, err)
```

Comment:
```suggestion
		return err
```

since the provider name is already in the returned error